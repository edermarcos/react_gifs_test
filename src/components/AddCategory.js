import React from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategory }) => {
  const [inputValue, setInputValue] = React.useState('');

  const handleSubmit = (e) => {
    e.preventDefault();

    if (!inputValue.trim()) {
      return;
    }

    /*
     * Se recibe como parametro el setter desde el padre
     * Por default, entites es el parametro que retorna y se crea un nuevo arreglo
     * */
    setCategory(entities => [inputValue, ...entities]);
    setInputValue('');
  };

  return (
    <form onSubmit={ handleSubmit }>
      <p>{ inputValue }</p>
      <input type="text"
             value={ inputValue }
             onChange={ (event) => setInputValue(event.target.value) }/>
    </form>
  );
};

AddCategory.propTypes = {
  setCategory: PropTypes.func.isRequired
};
