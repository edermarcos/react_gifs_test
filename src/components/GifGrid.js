import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category, deleteCategoryById }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <div>
      <div className="header">
        <h3>Category: { category }</h3>

        <div>
          <button onClick={ () => deleteCategoryById(category) }>Delete</button>
        </div>
      </div>

      { loading && <p>Loading...</p> }

      <div className="card-grid">
        {
          images.map(entity => (
            <GifGridItem key={ entity.id }
                         { ...entity }
            />
          ))
        }
      </div>
    </div>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
  deleteCategoryById: PropTypes.func,
};
