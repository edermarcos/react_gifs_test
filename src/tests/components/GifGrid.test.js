import { shallow } from 'enzyme';
import React from 'react';

import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Tests for the component <GifGrid />', () => {
  const category = 'One Punch';

  test('should render correctly', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={ category }/>);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
    const gifs = [
      {
        title: 'Item 1',
        url: 'https://localhost:8080.com',
        id: '1'
      }, {
        title: 'Item 2',
        url: 'https://localhost:8080.com',
        id: '2'
      }, {
        title: 'Item 3',
        url: 'https://localhost:8080.com',
        id: '3'
      }
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={ category } />)
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
