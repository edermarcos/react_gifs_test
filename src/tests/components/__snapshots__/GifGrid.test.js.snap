// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the component <GifGrid /> should render correctly 1`] = `
<div>
  <div
    className="header"
  >
    <h3>
      Category: 
      One Punch
    </h3>
    <div>
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </div>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</div>
`;
